components/NavLink.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import Link from 'next/link'
+
+type NavLinkProps = {
+  text: string
+  path: string
+  activePath?: string
+  external?: true
+  as?: string
+  blank?: true
+  className?: string
+}
+
+const NavLink: React.FC<NavLinkProps> = ({
+  text,
```

Comment:
I don't see the point of the `text` props . You could use `children` .
With `children` we have more freedom you can do:
```js
<NavLink>
  {children} // could be a text, img or a node
</NavLink>

<NavLink children="You text here if you want a self-closing tag" />
```

Reply:
didn't know this.... haha thanks @Wolfy64

Reply:
Makes sense :)

Reply:
fixed!